sources/api/thar-be-updates/src/status.rs

```diff
@@ -0,0 +1,395 @@
+use crate::error;
+use crate::error::Result;
+use bottlerocket_release::BottlerocketRelease;
+use chrono::{DateTime, Utc};
+use model::modeled_types::FriendlyVersion;
+use serde::{Deserialize, Serialize};
+use signpost::State;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::convert::TryInto;
+use std::os::unix::process::ExitStatusExt;
+use std::process::Output;
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum UpdateState {
+    Idle,
+    Available,
+    Staged,
+    Ready,
+}
+
+/// UpdateImage represents a Bottlerocket update image
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct UpdateImage {
+    arch: String,
+    version: semver::Version,
+    variant: String,
+}
+
+impl UpdateImage {
+    pub fn get_version(&self) -> &semver::Version {
+        &self.version
+    }
+}
+
+/// StagedImage represents an Bottlerocket image that is written to a partition set
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct StagedImage {
+    image: UpdateImage,
+    next_to_boot: bool,
+}
+
+impl StagedImage {
+    pub(crate) fn set_next_to_boot(&mut self, next_to_boot: bool) {
+        self.next_to_boot = next_to_boot
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum CommandStatus {
+    Success,
+    Failed,
+    Unknown,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum UpdateCommand {
+    RefreshUpdate,
+    PrepareUpdate,
+    ActivateUpdate,
+    DeactivateUpdate,
+    GetUpdateStatus,
+    GetAvailableUpdates,
+    GetUpdateInfoAbout,
+}
```

Comment:
nit: seems a little odd to define these here instead of in `main.rs`

Reply:
I need access to this enum from within the `status` module for storing the command information in the update status.